### PR TITLE
Role based authorization and UI changes

### PR DESCRIPTION
**Pull request by sreenidhi3**

### CODE REVIEW

File: src/layouts/sidebar/index.tsx [hunk 1]
```diff
@@ -100,13 +110,45 @@ export default function Sidebar() {
...
 
+  const departmentHeadAccessSidebar = [
+    ...employeeAccessSidebar,
+    {
+      label: 'Department',
+      icon: <DomainIcon fontSize='large' />,
+      path: ROUTE.DEPARTMENT,
+    },
+  ];
+
```
Comment (Surajrane45) [inline src/layouts/sidebar/index.tsx:132]: We can actually move all this sidebar related code to a separate file. Lets say getSidebarConfig.tsx
And move all this code there into a function which returns finally the sidebarConfig. We just need to pass the role to that function

File: src/layouts/sidebar/index.tsx [hunk 2]
```diff
@@ -140,14 +182,18 @@ export default function Sidebar() {
...
           >
-            <Button
-              variant='outlined'
-              sx={{ color: '#ffffff' }}
-              endIcon={<LogoutIcon />}
-              onClick={onLogout}
-            >
-              Logout
-            </Button>
+            {window.location.pathname === ROUTE.LOGIN ? (
```
Comment (Surajrane45) [inline src/layouts/sidebar/index.tsx:185]: Do we really need this condition? Sidebar should always mounted only for components where we need it. So if we don't need the sidebar we can avoid using the withLayout HOC for it

File: src/layouts/sidebar/index.tsx (hunk 1 above)
Comment (sreenidhi3) [reply src/layouts/sidebar/index.tsx:132]: okay

File: src/layouts/sidebar/index.tsx (hunk 2 above)
Comment (sreenidhi3) [reply src/layouts/sidebar/index.tsx:185]: okay

File: src/modules/Shared/HOC/privateRoute.tsx [hunk 3]
```diff
@@ -18,6 +20,16 @@ const PrivateRoute: React.FC<IProps> = ({ Component }) => {
     return null;
   }
 
+  //check if user has valid role to access the component
+  if (!Role.includes(auth.role)) {
+    if (auth.role === ROLES.EMPLOYEE) {
```
Comment (Surajrane45) [inline src/modules/Shared/HOC/privateRoute.tsx:25]: Can we have better condition handling here. Like if auth.role isn't present in our ROLE constants it should be always Unauthorized access. If we need some additional condition checking for Employee role we can have a separate if loop with comments which specify the need of checking it

File: src/modules/Shared/HOC/privateRoute.tsx [hunk 4]
```diff
@@ -18,6 +20,16 @@ const PrivateRoute: React.FC<IProps> = ({ Component }) => {
     return null;
   }
 
+  //check if user has valid role to access the component
+  if (!Role.includes(auth.role)) {
```
Comment (Surajrane45) [inline src/modules/Shared/HOC/privateRoute.tsx:24]: instead of Role we can use allowedRoles which sounds much better here

**Comment by sreenidhi3** [comment]
I have made the suggested changes. Could you please review?

**Review by Surajrane45 — APPROVED** [review]
LGTM

**Pull request merged**